### DISCUSSION TITLE
Wrapping a native object in a skia-safe object

### DISCUSSION THREAD

**Discussion by jdelliot** [rust-skia/rust-skia#635] [Q&A] [answered]
I am using skia-safe to do some integration with other systems. In some cases I will have and SkSurface object handed to me. Is it possible to wrap that object in a skia_safe::Surface? I might be overlooking something, but so far I haven't been able to find the hooks to make this happen. Seems like what I need is all pub(crate).

Thanks in advance, John

**Comment by pragmatrix** [comment, marked as answer]
Hello John, all rust-skia wrappers are supposed to implement matching traits in [`skia-safe/src/wrapper.rs`](https://github.com/rust-skia/rust-skia/blob/ee136359365c87617fdef1aa23bccc547b13ff8a/skia-safe/src/wrapper.rs). These are public and support the wrapping and unwrapping of Sk types.

**Reply by jdelliot** [reply]
@pragmatrix, thanks for your super quick response.
